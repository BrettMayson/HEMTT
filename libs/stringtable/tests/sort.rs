#![allow(clippy::unwrap_used)]

use hemtt_stringtable::Project;
use hemtt_workspace::WorkspacePath;

#[test]
fn sort_ace_arsenal() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/ace_arsenal.xml").unwrap()).unwrap();
    insta::assert_debug_snapshot!(stringtable);

    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);
}

#[test]
fn sort_comments() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/comments.xml").unwrap()).unwrap();
    insta::assert_debug_snapshot!(stringtable);

    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}

#[test]
fn sort_gh822() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/gh822.xml").unwrap()).unwrap();
    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}

#[test]
fn sort_containers() {
    let mut stringtable =
        Project::read(WorkspacePath::slim_file("tests/sort/containers.xml").unwrap()).unwrap();
    stringtable.sort();

    insta::assert_debug_snapshot!(stringtable);

    let mut out = String::new();
    stringtable.to_writer(&mut out, false).unwrap();

    insta::assert_snapshot!(out);
}
