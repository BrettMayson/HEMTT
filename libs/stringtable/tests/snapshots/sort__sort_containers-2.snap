---
source: libs/stringtable/tests/sort.rs
expression: out
---
<?xml version="1.0" encoding="utf-8"?>
<Project name="test">
    <Package name="test">
        <Key ID="A1">
            <Original>Origin</Original>
        </Key>
        <Container name="A">
            <Key ID="A3">
                <Original>Origin</Original>
            </Key>
        </Container>
        <Container name="B">
            <Key ID="A2">
                <English>EN</English>
                <German>DE</German>
            </Key>
            <Key ID="B1">
                <Original>Origin</Original>
            </Key>
        </Container>
    </Package>
</Project>
